src/components/sortable_table/header.js

```diff
@@ -23,78 +22,77 @@ var TableHeader = React.createClass({
     onLabelClick: React.PropTypes.func
   },
 
-  getInitialState: function() {
-    /*
-     * Keeps track of sort direction on each column.
-     */
-    var directionHash = _.zipObject(this.props.columns,
-      _.times(this.props.columns.length, function() { return 'ascending'; }));
+  /*
+   * Keeps track of sort direction on each column.
+   */
+  getInitialState() {
+    let directionHash = {};
+    this.props.columns.forEach((column) => {
+      directionHash[column] = 'ascending';
+    });
 
     return {
-      directionHash: directionHash
+      directionHash
     };
   },
 
-  onLabelClick: function(columnName, ev) {
-    /*
-     * Grabs table type and sort option and passes to sort callback.
-     * Flips direction in direction hash.
-     */
-    var direction = this.state.directionHash[columnName] === 'ascending' ? 'descending' : 'ascending';
-    var hashCopy = _.cloneDeep(this.state.directionHash);
+  /*
+   * Grabs table type and sort option and passes to sort callback.
+   * Flips direction in direction hash.
+   */
+  onLabelClick(columnName, event) {
+    let direction = this.state.directionHash[columnName] === 'ascending' ? 'descending' : 'ascending';
+    let updatedHash = assign({}, this.state.directionHash, {
+      [columnName]: direction
+    });
 
     this.props.onLabelClick(this.props.tableType, columnName, direction);
-    hashCopy[columnName] = direction;
-
     this.setState({
-      directionHash: hashCopy
+      directionHash: updatedHash
     });
   },
 
-  render: function() {
-    /*
-     * Render column labels and optionally render an expander element that proxies click events
-     * to table (Note: we'll only render this if we have a control column to render it into).
-     */
-    var hasProductColumn = _.contains(this.props.columns, 'product');
-
-    var control = this.props.isBulkEditable ?
-        <th key='control' className='sortable__label control' /> : null;
+  /*
+   * Render column labels and optionally render an expander element that proxies click events
+   * to table (Note: we'll only render this if we have a control column to render it into).
+   */
+  render() {
+    let props = this.props;
+    let control = props.isBulkEditable ?
```

Comment:
I'd also look at breaking control and labels out. I've often found it can be easier to follow when render is as thin as possible.